clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinatorTest.java

```diff
@@ -2589,6 +2557,54 @@ public void testConsumerRejoinAfterRebalance() {
         }
     }
 
+    @Test
```

Comment:
These two are the new tests to make sure when the flag is turned on, the behavior will be safe for supported version and panic when unsupported.